spectrum-offchain-lm/src/data/bundle.rs

```diff
@@ -140,24 +183,18 @@ impl IntoBoxCandidate for StakingBundle {
 impl TryFromBox for StakingBundle {
     fn try_from_box(bx: ErgoBox) -> Option<StakingBundle> {
         if let Some(ref tokens) = bx.tokens {
-            if tokens.len() == 2 && bx.ergo_tree == bundle_validator() {
+            if tokens.len() == 3 && bx.ergo_tree == bundle_validator() {
                 let redeemer_prop = ErgoTree::sigma_parse_bytes(
-                    &*bx.additional_registers
-                        .get(NonMandatoryRegisterId::R4)?
-                        .as_option_constant()?
+                    &bx.get_register(NonMandatoryRegisterId::R4.into())?
```

Comment:
Trying to work on `additional_registers` field directly crashes. Better to work with `get_register` method directly (`sigma-rust` should actually be updated here, make `additional_registers` non-pub).